kong/init.lua

```diff
@@ -123,6 +124,47 @@ local declarative_entities
 local schema_state
 
 
+local stash_init_worker_error
+local log_init_worker_errors
+do
+  local init_worker_errors
+  local init_worker_errors_str
+  local ctx_k = {}
+
+
+  stash_init_worker_error = function(err)
+    if err == nil then
+      return
+    end
+
+    err = tostring(err)
+
+    if not init_worker_errors then
+      init_worker_errors = {}
+    end
+
+    table.insert(init_worker_errors, err)
+    init_worker_errors_str = table.concat(init_worker_errors, ", ")
```

Comment:
Really minor suggestion: we can save this into `log_init_worker_errors` like:

```lua
    if not init_worker_errors or ngx.ctx[ctx_k] then
      return
    end

    if not init_worker_errors_str then
      init_worker_errors_str = table.concat(init_worker_errors, ", ")
    end
```

to avoid some unnecessary string creations. But a really minor point and certainly not a blocker.

Reply:
It's a tradeoff; the current approach saves the hot code path from executing `table.concat` entirely. Since `stash_init_worker_error` is only called during `init_worker`, we expect very few strings to be created at all, and they would be GC'ed. While the approach you suggested lazily calls `table.concat` on the proxy's hot code path, which is fine as well.

I'll leave the code as-is given both approaches are quite similar given the tradeoffs.